java/memory/src/main/java/org/apache/arrow/memory/NettyAllocationManager.java

```diff
@@ -34,31 +33,34 @@
   static final UnsafeDirectLittleEndian EMPTY = INNER_ALLOCATOR.empty;
   static final long CHUNK_SIZE = INNER_ALLOCATOR.getChunkSize();
 
-  private final int allocatedSize;
+  private final long allocatedSize;
   private final UnsafeDirectLittleEndian memoryChunk;
+  private final long allocatedAddress;
 
-  NettyAllocationManager(BaseAllocator accountingAllocator, int requestedSize) {
+  NettyAllocationManager(BaseAllocator accountingAllocator, long requestedSize) {
     super(accountingAllocator);
-    this.memoryChunk = INNER_ALLOCATOR.allocate(requestedSize);
-    this.allocatedSize = memoryChunk.capacity();
-  }
-
-  /**
-   * Get the underlying memory chunk managed by this AllocationManager.
-   * @return buffer
-   */
-  UnsafeDirectLittleEndian getMemoryChunk() {
-    return memoryChunk;
+    if (requestedSize > Integer.MAX_VALUE) {
+      memoryChunk = null;
+      allocatedAddress = PlatformDependent.allocateMemory(requestedSize);
+    } else {
+      this.memoryChunk = INNER_ALLOCATOR.allocate(requestedSize);
+      allocatedAddress = memoryChunk.memoryAddress();
```

Comment:
could you keep `allocatedSize = memoryChunk.capacity()` for this case?

Reply:
Sure. Revised. Please check.